### DISCUSSION TITLE
Unable to push a view

### DISCUSSION THREAD

**Discussion by micbakos** [rundfunk47/stinsen#55] [Q&A] [answered]
Hello,

Thank you for this amazing library. I am an Android Developer, new to iOS, developing an app in both platforms with Kotlin Multiplatform. 

I am going to describe my use case in some general terms:
The "shared" codebase contains `ViewModels`, which keep and mutate the view state and when it comes to navigation, emit `NavigationCommands`. These commands are platform agnostic. Meaning, they represent a screen that the app needs to navigate to, while being declared and used in the shared codebase. Now each platform (Android, iOS) listens to these commands and navigates accordingly to the correct destination. When it comes to iOS and SwiftUI I discovered this library as an imperative approach to routing decoupled from the view system.

Now let's say that I have two screens that I need to display to the user. `FirstScreen` and `SecondScreen`.

I declare a coordinator like this:
```
final class MainCoordinator: NavigationCoordinatable {
    let stack: NavigationStack<MainCoordinator> = NavigationStack(initial: \.first)
    
    @Root var first = makeFirst
    @Route(.push) var second = makeSecond
    
    @ViewBuilder func makeFirst() -> some View {
        FirstScreen()  // `struct FirstScreen: View { ... }`
    }
    
    @ViewBuilder func makeSecond() -> some View {
        SecondScreen()
    }
}
``` 

And listen to the navigation commands in the app 
```
private let coordinator: MainCoordinator = MainCoordinator()
var body: some Scene {
    WindowGroup {
         coordinator
             .view()
             .onAppear(perform: {
                 // Injecting the navigator instance, from the shared framework, that listens to the navigation commands
                 SharedApplication().injectNavigator().onCommandReceived(onCommand: { command in 
                     // Now I can translate a command to a route, i.e.
                     if navigationCommand is GoToSecondScreen {
                          coordinator.route(to: \.second)
                     }
                 })
             })
     }
}
```

This doesn't result in any navigation. On the other hand, if I change the `@Route(.push)` to `Route(.modal)` the navigation works, and the second screen is presented as a modal.

Now the question is:
1. Do I declare the second route wrong? I studied the documentation and I saw an example of a push route being declared as a `NavigationCoordinatable` and not a `ViewBuilder`. On the other hand, the sample app declares multiple push routes as `ViewBuilder`s, so that doesn't seem to be the case.
2. Do I need to make changes to the views First and Second screen views themselves, in order to accomodate any push navigation?

I would appreciate any help. Thank you!

**Comment by YuantongL** [comment, marked as answer]
Hi, did not run to test, but my take is that your view is not in a navigation stack, so push won't work.
Can you try warp your coordinator into a `NavigationViewCoordinator` like
```
NavigationViewCoordinator(coordinator)
.view()
```

**Reply by micbakos** [reply]
Yeah found out that too, turns out, I am not familiar with the ios navigation system yet 😛 . Thank you!
